common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala

```diff
@@ -140,6 +140,19 @@ class RollupManager(pgu: PGSecondaryUniverse[SoQLType, SoQLValue], copyInfo: Cop
     }
   }
 
+  def bumpRollupVersionByCopyTable(rollupInfo: RollupInfo, oldCopyInfo: Option[CopyInfo], newDataVersion: Long): Unit = {
+    val newTableName = rollupTableName(rollupInfo, newDataVersion)
+    oldCopyInfo.filter(_.copyNumber == rollupInfo.copyInfo.copyNumber).map { ci =>
+      val oldTableName = rollupTableName(rollupInfo, ci.dataVersion)
+      using(pgu.conn.createStatement()) { stmt =>
+        val sql = s"CREATE TABLE ${newTableName} (LIKE ${oldTableName} including all); INSERT INTO ${newTableName} SELECT * FROM ${oldTableName};"
```

Comment:
I am a bit concerned this may be slower for large rollups as it inserts data into a table that is already indexed instead of inserting the data then creating indexes.  In the past, what I've seen is that for large rollups inserting the data and creating the indexes are 90%+ of the time and the actual query is <10%.

Have you looked at how hard it would be to create it without indexes then run the codepath that creates the indexes?  I'm not sure how flexible that code is.

This codepath only gets hit for rollup creation / deletion without data changes, so maybe that is an ok tradeoff for now?  At the least it would be good to have a comment here documenting that tradeoff.

Reply:
Also, note that this will result with indexes being named differently on the rollup.  I don't think that anything cares about that but wanted to call it out.

I also did a test just to satisfy myself there are no exclusive locks required on the source table.  I know there _shouldn't_ logically be but have run into somewhat related things in the past so wanted to be sure.

session 1:

```
[local]:5432 marcslemko@marcslemko=# begin;
BEGIN
[local]:5432 marcslemko@marcslemko=#* select * from foo;
 bar 
-----
   1
(1 row)

[local]:5432 marcslemko@marcslemko=#* \d foo
                Table "public.foo"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 bar    | integer |           |          | 
Indexes:
    "marcs1" btree (bar)
    "marcs2" btree (bar)
```
(transaction left open)

session 2:

```
[local]:5432 marcslemko@marcslemko=# begin;
BEGIN
[local]:5432 marcslemko@marcslemko=#* create table foo2 (like foo including all);
CREATE TABLE
[local]:5432 marcslemko@marcslemko=#* insert into foo2 select * from foo;
INSERT 0 1
[local]:5432 marcslemko@marcslemko=#* commit;
COMMIT
[local]:5432 marcslemko@marcslemko=# \d foo2
                Table "public.foo2"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 bar    | integer |           |          | 
Indexes:
    "foo2_bar_idx" btree (bar)
    "foo2_bar_idx1" btree (bar)
```